### ISSUE TITLE
Dont depend on sessions for "genealabs-laravel-model-caching-is-disabled"

### ISSUE THREAD

**Issue by andrewmclagan** [GeneaLabs/laravel-model-caching#75]
# Issue

Some users have installations that do not use sessions as they have stateless APIs that authenticate with JWT tokens (Laravel Passport). This architecture is common and popular as it adheres to the 12 factor app principals of stateless server apps.

We therefore disable session service providers and facades as there is a significant performance increase in doing so. Along with other services.

## Environment
Laravel Version: *5.5.x*
Laravel Model Caching Package Version: *latest*
PHP Version: *7.2.x*
Homestead Version: *N/A - Docker (docker makes homestead development look like 1965)*
Operating System & Version: *Alpine Linux 3.5*

## Stack Trace
```
26) Tests\Integration\Services\JobWorkflow\WorkflowDraftToPendingTest::it_throws_exception_on_insufficient_credits
Failed asserting that exception of type "ReflectionException" matches expected exception "App\Exceptions\InsufficientCredits". Message was: "Class session does not exist" at
/drone/src/bitbucket.org/ethical-jobs/api-app/vendor/laravel/framework/src/Illuminate/Container/Container.php:752
/drone/src/bitbucket.org/ethical-jobs/api-app/vendor/laravel/framework/src/Illuminate/Container/Container.php:631
/drone/src/bitbucket.org/ethical-jobs/api-app/vendor/laravel/framework/src/Illuminate/Container/Container.php:586
/drone/src/bitbucket.org/ethical-jobs/api-app/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:732
/drone/src/bitbucket.org/ethical-jobs/api-app/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php:110
/drone/src/bitbucket.org/ethical-jobs/api-app/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php:839
/drone/src/bitbucket.org/ethical-jobs/api-app/vendor/genealabs/laravel-model-caching/src/CachedModel.php:20
/drone/src/bitbucket.org/ethical-jobs/api-app/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:873
/drone/src/bitbucket.org/ethical-jobs/api-app/vendor/laravel/framework/src/Illuminate/Database/Eloquent/FactoryBuilder.php:169
/drone/src/bitbucket.org/ethical-jobs/api-app/vendor/laravel/framework/src/Illuminate/Support/Collection.php:339
/drone/src/bitbucket.org/ethical-jobs/api-app/vendor/laravel/framework/src/Illuminate/Database/Eloquent/FactoryBuilder.php:173
/drone/src/bitbucket.org/ethical-jobs/api-app/vendor/laravel/framework/src/Illuminate/Database/Eloquent/FactoryBuilder.php:153
/drone/src/bitbucket.org/ethical-jobs/api-app/tests/TestFactories/OrganisationFactory.php:37
/drone/src/bitbucket.org/ethical-jobs/api-app/tests/Integration/Services/JobWorkflow/WorkflowDraftToPendingTest.php:188
```

# Solution

In your class `GeneaLabs\LaravelModelCaching\CachedModel` you have a method `newEloquentBuilder()` that creates the builder instance. You are using a session based store to retain state regarding enabled / disabled status of the package.

As your package is based around caching one can expect a caching engine to be a requirement, the docs even suggest Redis as a best fit - with this I agree.

I believe its wrong to alos depend on the session driver for this package as another dependancy when you could simply store this `session('genealabs-laravel-model-caching-is-disabled')` key in the cache as well. Thus only having one external service dependancy - packages should have as few as possible to remain flexible. Also I cant see why session is being used for this over cache in the first place?

**Comment by andrewmclagan** [comment]
Id also change the key naming scheme to support the redis / industry standard naming format for cache keys `:` for example it give oyu namespacing powers in the key relm:

`genealabs-laravel-model-caching-is-disabled`

could be better written

`genealabs:model-caching:is-disabled`

thus your other keys can use the namespace

`genealabs:model-caching:`

**Comment by mikebronner** [comment]
@andrewmclagan both great suggestions! I completely agree with the sessions being a pain in the butt. Will work on integrating these ideas, thanks!

**Comment by mikebronner** [comment]
Fixed in PR #76
